### ISSUE TITLE
 Unlock password dialog sometimes lacks keyboard focus

### ISSUE THREAD

**Issue by raveit65** [mate-desktop/mate-screensaver#82]
origin downstream report at fedora bugzilla:
https://bugzilla.redhat.com/show_bug.cgi?id=984333#c16
```
 David Gibson 2013-07-14 21:01:17 EDT

Description of problem:

Sometimes when returning to a MATE session when the screensaver has kicked in, the password box to unlock the screen does not have keyboard focus, and so cannot be completed.  The only way to proceed is to kill mate-screensaver from a VT.

Version-Release number of selected component (if applicable):

mate-screensaver-1.6.0-1.fc19.x86_64

(also observed in Fedora 18)

How reproducible:

Intermittent, no obvious pattern so far.

Steps to Reproduce:
1. Configure screensaver to lock with password
2. Allow screensaver to start
3. Attempt to unlock screen with password

Actual results:

Sometimes, although pressing a key does bring up the password prompt, the prompt does not appear to have keyboard focus and nothing appears when attempting to type password.  Other times it works as expected.

If this happens once, allowing the password prompt to time out (or pressing Cancel with the mouse) then retriggering the password prompt will suffer the same problem.

Expected results:

Password box works, and unlocks screen.

Additional info:

Suspect this may be related to some other windows on the locked desktop which attempt to grab all focus (e.g. VM console from virt-manager) but I don't have confirmation or a clear reproducer yet.
```
confirmation with 1.10 in fedora 22:
```
 David Gibson 2015-09-24 21:36:04 EDT

Yes, I'm still seeing this quite often with Fedora 22 and mate-screensaver-1.10.2-1.fc22.x86_64.  I'd say it happens maybe 60-70% of the times the screensaver activates.
```
Note, this is in xnomad session, i will try to reproduce it

**Comment by harinath** [comment]
There's also a Debian bug with the same symptoms:

  https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=785739

**Comment by raveit65** [comment]
I could reproduce the issue if xmonad WM is running on my notebook.
see. https://bugzilla.redhat.com/show_bug.cgi?id=984333#c23
My logs shows that fprintd daemon starts without reason on a system w/o a hardware for a fingerprint reader and the lookscreen is set to sensitive, which is the correct behaviour if a fingerprint reader is installed.
https://github.com/mate-desktop/mate-screensaver/blob/master/src/gs-lock-plug.c#L1606
After disable the service i couldn't reproduce the issue anymore.
On the other side the reporter says fprintd isn't installed.
https://bugzilla.redhat.com/show_bug.cgi?id=984333#c26
So there must be another service too which forced mate-screensaver to set the screen to sensitive.

**Comment by physicsshark** [comment]
I can confirm that this is not xmonad specific.  I have the same issue (with Compiz).  However, I cannot blame it on fprintd, because I do not have fprintd installed.  
I cannot emphasize enough how much this bug bothers me.
So far, I've had two different solutions to fix this when it happens:
1. Kill xserver (using keyboard shortcut), but this sometimes causes me to lose work.
2. Log into a TTY to specifically kill mate-screensaver.  Then, I restart it when I'm back into my session.

Any workarounds that are less intrusive would be much appreciated.

**Comment by raveit65** [comment]
see https://github.com/mate-desktop/mate-screensaver/issues/73#issuecomment-120256363
Maybe this workaround help.

**Comment by physicsshark** [comment]
That workaround does help, but not every display manager is supported.  I think you have to use LightDM for this workaround to work.  I do use LightDM, but sometimes for whatever reason the switch user button is missing from the unlock dialog.  This is when I have to resort to more hacky workarounds.

**Comment by raveit65** [comment]
yes, this was with lightdm
I don't know which distro you're using and how mate-screensaver is commpiled there.
Did your distro use systemd and not consolekit?
I stumbled about a weird configure flag issue with m-s.
M-s can compiled with --with-systemd but consolekit is always enabled if you don't explicit use --without-console-kit.
I run is this configure mismatch in my fedora packages, maybe this is the cause of the issue.

**Comment by physicsshark** [comment]
I'm on Arch, which uses systemd.  I do not have consolekit installed.  I'm not familiar with how mate-screensaver is compiled on Arch or otherwise.

**Comment by physicsshark** [comment]
I just looked at the Arch PKGBUILD.  mate-screensaver is not compiled with the --without-console-kit option on Arch.
https://projects.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/mate-screensaver-gtk3

**Comment by raveit65** [comment]
Well, this is definitive wrong, same as in fedora.
Because than code is used which isn't for systemd systems.
I will rebuild m-s for fedora with --without-console-kit option an ask reporter if this make a different.

**Comment by flexiondotorg** [comment]
I'll rebuild the Arch package using `--without-console-kit` too.

**Comment by flexiondotorg** [comment]
Updated package released to the Arch Linux [community] repo. @physicsshark please test and report back :-)

**Comment by physicsshark** [comment]
@flexiondotorg Will do.  :-)

**Comment by physicsshark** [comment]
@flexiondotorg I'm not sure what compiling with this option was supposed to do.  I've already had one incident where unlocking didn't work after upgrading to the newer version. Or was this supposed to fix the "Switch user" option from disappearing?

**Comment by physicsshark** [comment]
@raveit65 After reading through my journalctl logs after each time this happens, I'm beginning to think that this has something to do with NetworkManager (maybe NetworkManager is stealing keyboard focus).  This would explain why this issue often happens when my computer is resuming from suspend or hibernate, because NetworkManager has to reconnect.

**Comment by physicsshark** [comment]
@raveit65 @flexiondotorg Turns out that NetworkManager had nothing to do with it.  This problem persisted even when I used other network management utilities.  Fortunately, I had the chance to extensive testing over the holidays to find out which application/configuration was causing the problem.  And, the guilty party is . . . conky, everyone's favorite ultra-customizable system monitor.  Here's my reasoning:

| Conky        | The Issue           |
| ------------- |:-------------:|
| Some people have it, some people don't | Some people have it, some people don't |
| When no applications are open, conky often steals keyboard focus so that keyboard shortcuts don't work | Sounds like a similar issue|
| When conky is not running, or my hack (described below) is used... | The issue doesn't occur|
| When conky is running without my hack... | The issue often occurs |

Users have reported having this issue with xmonad, which makes sense to me because many users that use minimalist window managers use conky.  This issue was occurring for me with compiz. 
Here's my workaround for compiz:
![screenshot at 2016-01-08 22 04 44](https://cloud.githubusercontent.com/assets/9755486/12214041/e05c7c5a-b653-11e5-9fb0-fcb081057baf.png)
Most window managers have some way to control window rules in a similar way to compiz, and I would assume xmonad would as well.

I'm not trying to claim that conky is the only cause for this issue, but conky was **definitely** the cause for me.

I hope this helps.  :smiley_cat:

**Comment by raveit65** [comment]
Thanks for the info.
To sumary the issue:
Several application can steal the focus of the password-dialog.

**Comment by zensubz** [comment]
@physicsshark Thanks very much for your inputs. I have been facing this problem, and yes indeed, as you pointed out, it is Conky that's responsible for this issue (in my case as well). Don't know if there are any other progs that might cause this issue. 

All I did was to check screenlock by keeping Conky running, and I couldn't gain the keyboard focus. Had to use tty to kill m-s... In the second method I ensured Conky wasn't running, then screen-locked my desktop, and this time there was no problem of keyboard losing the focus... 

I investigated the Conky variable that might cause this issue: In my case (with Conky 1.9.x), it was actually this one: **own_window_type normal**. Changing it to **own_window_type override** solved this issue concerning Conky and m-s, and this time there was no problem of losing the keyboard-focus for unlocking...

**Comment by jasminhassan** [comment]
this was happening to me in Ubuntu 17.10 and now this happened to me again in Ubuntu 18.04 (developer release) with MATE 1.20
Also running Conky (1.10.8). Will try the suggestion by zensubz above.

**Comment by lukefromdc** [comment]
This sounds like someone needs to open a bug report against conky.
